agate/agate/tests/test_queue_reading.py

```diff
@@ -0,0 +1,154 @@
+            "gzip":"gz"
+         }
+      },
+      ".2.fastq.gz":{
+         "uri":"s3://synthscape-synthscape.ukhsa-illumina-test/synthscape.0.92274db4-3535-47db-b6f6-151354597142.2.fastq.gz",
+         "etag":"de2fed484dcad68d50ec6673df6b3670",
+         "key":"synthscape.0.92274db4-3535-47db-b6f6-151354597142.2.fastq.gz",
+         "submitter":"bryn-synthscape-ukhsa",
+         "parsed_fname":{
+            "project":"synthscape",
+            "run_index":"0",
+            "run_id":"92274db4-3535-47db-b6f6-151354597142",
+            "direction":"2",
+            "ftype":"fastq",
+            "gzip":"gz"
+         }
+      },
+      ".csv":{
+         "uri":"s3://synthscape-synthscape.ukhsa-illumina-test/synthscape.0.92274db4-3535-47db-b6f6-151354597142.csv",
+         "etag":"714c18df9c62b30f70443219467fcba1",
+         "key":"synthscape.0.92274db4-3535-47db-b6f6-151354597142.csv",
+         "submitter":"bryn-synthscape-ukhsa",
+         "parsed_fname":{
+            "project":"synthscape",
+            "run_index":"0",
+            "run_id":"92274db4-3535-47db-b6f6-151354597142",
+            "ftype":"csv"
+         }
+      }
+   },
+   "test_flag":true
+}"""
+
+
+class QueueReadingTestCase(TestCase):
+
+    def test_calling_inbound_matched(self):
+
+        self.assertEqual(IngestionAttempt.objects.count(), 0)
+
+        retrieval = iMessageRetrieval()
+        retrieval.receive_batch = Mock(return_value=[])
+
+        q = QueueReader(retrieval)
+        q.update()
+
+        retrieval.receive_batch.assert_any_call(exchange="inbound-matched")
+
+    def test_inbound_matched_project_and_site_updates(self):
+
+        self.assertEqual(IngestionAttempt.objects.count(), 0)
+        self.assertEqual(Project.objects.count(), 0)
+        self.assertEqual(ProjectSite.objects.count(), 0)
+
+        message = Mock()
+        message.body = inbound_matched_example
+
+        retrieval = iMessageRetrieval()
+
+        def side_effect_func(exchange):
+            if exchange == "inbound-matched":
+                return [message]
+            else:
+                return []
+        retrieval.receive_batch = Mock(side_effect=side_effect_func)
+        retrieval.acknowledge_message = Mock()
+
+        q = QueueReader(retrieval)
+        q.update()
+
+        retrieval.receive_batch.assert_any_call(exchange="inbound-matched")
+        retrieval.acknowledge_message.assert_any_call(message)
+
+        self.assertEqual(IngestionAttempt.objects.count(), 1)
+        self.assertEqual(Project.objects.count(), 1)
+        self.assertEqual(ProjectSite.objects.count(), 1)
+        Project.objects.get(key="synthscape")
+        ProjectSite.objects.get(key="synthscape-bham")
+
+        retrieval.receive_batch.assert_any_call(
+            exchange="inbound-to_validate-synthscape")
+        retrieval.receive_batch.assert_any_call(
+            exchange="inbound-results-synthscape-bham")
+
+    def test_inbound_matched_injestion_attempt(self):
```

Comment:
Why is this a separate test? It looks like we're just testing more properties of the same message (after we retrieve it by `uuid`).

Reply:
The asserts after following the method are different. 

- Testing that the inbound message updates the list of project-sites, and the new projects/sites have their queues queried
- Testing that inbound messages cause updates to the database of ingestion attempts

They are testing 2 different consequences (of the same code) and so I believe they should be in 2 differently named unit tests (so that one failing test gives you enough information to be able to diagnose one failure mode).